File: backend/app/schemas/matching.py
from uuid import UUID

from pydantic import BaseModel

from app.schemas.common import APIModel

class MatchQueryInput(BaseModel):
    query_text: str


class MatchedExpertSummary(APIModel):
    expert_id: UUID
    full_name: str
    email: str
    short_bio: str | None = None
    aggregate_similarity_score: float
    matched_document_excerpt: str
    website_url: str | None = None
    x_handle: str | None = None
    linkedin_identifier: str | None = None
    bluesky_identifier: str | None = None
    github_handle: str | None = None


class MatchQueryResponse(APIModel):
    match_query_id: UUID
    applied_match_acceptance_threshold: float
    matches: list[MatchedExpertSummary]

File: frontend/src/services/matching.ts
import { api } from "./api";

export type MatchRequest = {
  query_text: string;
};

export type MatchedExpert = {
  expert_id: string;
  full_name: string;
  email: string;
  short_bio?: string;
  aggregate_similarity_score: number;
  matched_document_excerpt: string;
  website_url?: string;
  x_handle?: string;
  linkedin_identifier?: string;
  bluesky_identifier?: string;
  github_handle?: string;
};

export type MatchResponse = {
  match_query_id: string;
  applied_match_acceptance_threshold: number;
  matches: MatchedExpert[];
};

export const matchingApi = {
  createQuery: (payload: MatchRequest) => api.post<MatchResponse>("/api/v1/match-queries", payload),
};
